lib/entities/semantic-file.js

```diff
@@ -0,0 +1,161 @@
+import { sparqlEscape } from 'mu';
+import { SERVICE_URI } from '../../env';
+import { filenameToUri, uriToFilename } from '../util/file';
+import { v5 as uuidv5 } from 'uuid';
+import path from 'path';
+import fs from 'fs-extra';
+import Base64 from 'crypto-js/enc-base64';
+import sha256 from 'crypto-js/sha256';
+import moment from 'moment';
+
+const NAMESPACE = '77219b1e-e23d-4828-847f-55e2d7fac687'; // NOTE: used for consistent uuid generation based on the URI
```

Comment:
Consistent on length ?

Reply:
No, it is name-based, see [uuidv5](https://www.terraform.io/docs/language/functions/uuidv5.html) for more info. But basically it guaranties that based on the filename/URI it generates always the same UUID for that specific file.